File: app/src/main/java/org/record/tiny/demo/main/MainContract.java
package org.record.tiny.demo.main;

import org.record.tiny.base.BaseView;
import org.record.tiny.demo.model.Tab;

import java.util.List;

@SuppressWarnings("All")
public class MainContract {
    interface View extends BaseView {
        void getTabs(List<Tab.DataBean> tabs);

        void showLoading();

        void hideLoading();
    }

    interface Presenter {
        void checkPermissions(boolean success);

        void login(boolean success);

        void getTabs(boolean success, List<Tab.DataBean> tabs);
    }
}

File: app/src/main/java/org/tiny/record/view/main/MainContract.kt
package org.tiny.record.view.main

import org.tiny.lib.core.BaseView

/**
 * Created by tiny on 2/11/2018
 */
class MainContract {
    interface IView : BaseView {
        fun jumpEditPage()
        fun jumpPreviewPage(id: Int)
    }
}